### PR TITLE
feat: build nearline tablet framework 

### LINKED ISSUE #218: Build NearLineTablet framework
Add code to java/openmldb-tablet

+ use [brpc-java](https://github.com/baidu/brpc-java)
+ create table

### PR DESCRIPTION
**Pull request by dl239**
+ add nearline tablet module using brpc-java
+ create iceberg table

### CODE REVIEW

File: java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            builder.setCode(-1).setMsg("create table failed");
+        }
+        NLTablet.CreateTableResponse response = builder.build();
+        return response;
+    }
+
+    public boolean createTable(String dbName, String tableName, String partitionKey, List<ColumnDesc> schema) {
+        TableIdentifier table = TableIdentifier.of(dbName, tableName);
+        if (catalog.tableExists(table)) {
+            log.warn("table {} already exists in db {}", tableName, dbName);
```
Comment (imotai) [inline java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java]: return this error message to client

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    public boolean connectZookeeper() throws Exception {
+        ZKConfig config = ZKConfig.builder()
+                .cluster(NLTabletConfig.ZK_CLUSTER)
+                .namespace(NLTabletConfig.ZK_ROOTPATH)
+                .sessionTimeout(NLTabletConfig.ZK_SESSION_TIMEOUT)
+                .build();
+        zkClient = new ZKClient(config);
+        if (!zkClient.connect()) {
+            log.error("fail to connect zookeeper");
```
Comment (imotai) [inline java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java]: add zk endpoints to the error message

File: java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                .day(partitionKey)  // use day partition default
+                .build();
+        catalog.createTable(table, icebergSchema, spec);
+        log.info("create table {} success", tableName);
+    }
+
+    public Schema convertSchema(List<ColumnDesc> schema) throws Exception {
+        List<Types.NestedField> columns = new ArrayList<Types.NestedField>();
+        try {
+        for (int i = 0; i < schema.size(); i++) {
```
Comment (imotai) [inline java/openmldb-nearline-tablet/src/main/java/com/_4paradigm/openmldb/server/impl/NLTabletServerImpl.java]: pay attention to the code style

**Review by imotai — APPROVED** [review]
lgtm

File: java/openmldb-nearline-tablet/src/main/resources/log4j2.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+appenders=console
+appender.console.type=Console
+appender.console.name=console
+appender.console.layout.type=PatternLayout
+appender.console.layout.pattern=[%d{yy-MMM-dd HH:mm:ss:SSS}] [%p] [%c{1}:%L] - %m%n
+rootLogger.level=info
+rootLogger.appenderRefs=console_ref
+rootLogger.appenderRef.console_ref.ref=console
```
Comment (vagetablechicken) [inline java/openmldb-nearline-tablet/src/main/resources/log4j2.properties:8]: end of line

**Pull request merged**